crates/languages/src/python.rs

```diff
@@ -182,21 +182,94 @@ async fn get_cached_server_binary(
     }
 }
 
-pub(super) fn python_task_context() -> ContextProviderWithTasks {
-    ContextProviderWithTasks::new(TaskTemplates(vec![
-        TaskTemplate {
-            label: "execute selection".to_owned(),
-            command: "python3".to_owned(),
-            args: vec!["-c".to_owned(), VariableName::SelectedText.template_value()],
-            ..TaskTemplate::default()
-        },
-        TaskTemplate {
-            label: format!("run '{}'", VariableName::File.template_value()),
-            command: "python3".to_owned(),
-            args: vec![VariableName::File.template_value()],
-            ..TaskTemplate::default()
-        },
-    ]))
+pub(crate) struct PythonContextProvider;
+
+const PYTHON_UNITTEST_TARGET_TASK_VARIABLE: VariableName =
+    VariableName::Custom(Cow::Borrowed("PYTHON_UNITTEST_TARGET"));
+
+impl ContextProvider for PythonContextProvider {
+    fn build_context(
+        &self,
+        _variables: &task::TaskVariables,
+        _location: &project::Location,
+        _cx: &mut gpui::AppContext,
+    ) -> Result<task::TaskVariables> {
+        let _python_module_name = python_module_name_from_relative_path(
+            _variables.get(&VariableName::RelativeFile).unwrap_or(""),
+        );
+        let _unittest_class_name =
+            _variables.get(&VariableName::Custom(Cow::Borrowed("_unittest_class_name")));
+        let _unittest_method_name = _variables.get(&VariableName::Custom(Cow::Borrowed(
+            "_unittest_method_name",
+        )));
+
+        let _unittest_target_str = match (_unittest_class_name, _unittest_method_name) {
+            (Some(class_name), Some(method_name)) => {
+                format!("{}.{}.{}", _python_module_name, class_name, method_name)
+            }
+            (Some(class_name), None) => format!("{}.{}", _python_module_name, class_name),
+            (None, None) => format!("{}", _python_module_name),
```

Comment:
```suggestion
            (None, None) => _python_module_name,
```

Reply:
Thanks, you're so fast. Probably should've looked at clippy before pushing, fixing this and more now :)